samples/cpp/windows/console/samples/speech_recognition_samples.cpp

```diff
@@ -613,6 +615,68 @@ void PronunciationAssessmentWithMicrophone()
     }
 }
 
+void PronunciationAssessmentWithStreamInternalAsync(shared_ptr<SpeechConfig> speechConfig, std::string referenceText, std::vector<uint8_t> audioData, std::promise<int> resultReceived, std::vector<std::string>& resultContainer)
+{
+    auto audioFormat = AudioStreamFormat::GetWaveFormatPCM(16000, 16, 1); // This need be set based on the format of the given audio data
+    auto audioInputStream = AudioInputStream::CreatePushStream(audioFormat);
+    auto audioConfig = AudioConfig::FromStreamInput(audioInputStream);
+    auto speechRecognizer = SpeechRecognizer::FromConfig(speechConfig, audioConfig);
+
+    // create pronunciation assessment config, set grading system, granularity and if enable miscue based on your requirement.
+    auto pronAssessmentConfig = PronunciationAssessmentConfig::Create(referenceText, PronunciationAssessmentGradingSystem::HundredMark, PronunciationAssessmentGranularity::Phoneme, false);
+    pronAssessmentConfig->ApplyTo(speechRecognizer);
+
+    audioInputStream->Write(audioData.data(), static_cast<uint32_t>(audioData.size()));
+    audioInputStream->Write(nullptr, 0);// send a zero-size chunk to signal the end of stream
+
+    auto result = speechRecognizer->RecognizeOnceAsync().get();
+    if (result->Reason == ResultReason::Canceled)
+    {
+        auto cancellationDetails = CancellationDetails::FromResult(result);
+        std::cout << cancellationDetails->ErrorDetails << std::endl;
+    }
+    else
+    {
+        auto responseJson = result->Properties.GetProperty(PropertyId::SpeechServiceResponse_JsonResult, "");
+        resultContainer.push_back(responseJson);
+    }
+
+    resultReceived.set_value(1);
+}
+
+// Pronunciation assessment with audio stream input.
+// See more information at https://aka.ms/csspeech/pa
+void PronunciationAssessmentWithStream()
+{
+    // Creates an instance of a speech config with specified subscription key and service region.
+    // Replace with your own subscription key and service region (e.g., "westus").
+    auto config = SpeechConfig::FromSubscription("YourSubscriptionKey", "YourServiceRegion");
+
+    // Read audio data from file. In real scenario this can be from memory or network
+    std::ifstream file("whatstheweatherlike.wav", std::ios::binary | std::ios::ate);
+    auto audioDataWithHeaderSize = file.tellg();
+    file.seekg(46);
+    auto audioData = std::vector<uint8_t>(static_cast<size_t>(audioDataWithHeaderSize) - static_cast<size_t>(46));
+
+    file.read((char*)audioData.data(), audioData.size());
+
+    std::promise<int> resultReceived;
```

Comment:
without using it?

Reply:
adding resultReceived.get_future() and futureResult.get()